homeassistant/components/nordpool/sensor.py

```diff
@@ -54,10 +54,11 @@ def get_prices(
     current_price_entries: dict[str, float] = {}
     next_price_entries: dict[str, float] = {}
     current_time = dt_util.utcnow()
-    previous_time = current_time - timedelta(hours=1)
-    next_time = current_time + timedelta(hours=1)
     LOGGER.debug("Price data: %s", data)
     for entry in data:
+        resolution = entry.end - entry.start
```

Comment:
- all intervals have the same resolution, right? if so, can we move this out of the loop and use the first 2 entries?
- is it guaranteed that the entries are sorted chronologically? should we add some check (or sort them), since they come from an external library that we don't have tests for (in here)?

Reply:
- The data can contain multiple markets each with their own resolution so while not likely, one can have 15 minutes and another 60 minutes.
- Doesn't matter since we loop all entries and only save and return the ones which fits the asked time window. That in itself is wasteful but would be for another PR to fix.